lib/Logger.js

```diff
@@ -20,72 +29,83 @@ class Logger {
     static get LogLevel() {
         return Object.keys(LogLevels).find(key => LogLevels[key] === LogLevel);
     }
+    static get LogLevels() {
+        return LogLevels;
+    }
     static set LogLevel(value) {
         if (LogLevels[value] === undefined) {
             throw "invalid log level '" + value + "', valid are '" + Object.keys(LogLevels).join("','") + "'";
         }
         LogLevel = LogLevels[value];
     }
+    static set LogFile(filename) {
+        if (LogFilename === filename) {
+            return;
+        }
+        if (LogFile) {
+            LogFile.close();
+        }
+        LogFilename = filename;
+        LogFile = fs.createWriteStream(LogFilename, LogFileOptions);
+    }
+    static get LogFile() {
+        return LogFilename;
+    }
+
+    static log(level, ...args) {
+        if (LogLevel <= LogLevels[level]) {
+            const logPrefix = BuildPrefix(level.toUpperCase());
+            // eslint-disable-next-line no-console
+            console.log(logPrefix, ...args);
+            LogFile.write([logPrefix, ...args].map(arg => {
+                if (typeof arg === "string") {
+                    return arg;
+                }
+                return util.inspect(arg);
+            }).join(" "));
+            LogFile.write("\n");
+        }
```

Comment:
While we're at it, it might also make sense to think about logging to mqtt maybe?

Reply:
Sure. Added that, not an MQTT-Expert though, so please have a close look.